camligorkem/assignment2/BinaryTree.java

```diff
@@ -0,0 +1,164 @@
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Optional;
+
+public class BinaryTree<T> {
+    private BinaryNode<T> root= null;
```

Comment:
Again, you don't need to initialise it to null

Reply:
done!